packager/util/util_test.go

```diff
@@ -27,3 +35,41 @@ func TestCanSignHttpExchanges(t *testing.T) {
 	// CA node does not.
 	assert.False(t, util.CanSignHttpExchanges(pkgt.Certs[1]))
 }
+
+func TestParseCertificate(t *testing.T) {
+	assert.Nil(t, util.CheckCertificate(pkgt.B3Certs[0], pkgt.B3Key, "amppackageexample.com"))
+}
+
+func TestParseCertificateSubjectAltName(t *testing.T) {
+	assert.Nil(t, util.CheckCertificate(pkgt.B3Certs[0], pkgt.B3Key, "www.amppackageexample.com"))
+}
+
+func TestParseCertificateNotMatchX(t *testing.T) {
+	assert.Contains(t, errorFrom(util.CheckCertificate(pkgt.B3Certs[0],
+		pkgt.B3Key2, "amppackageexample.com")), "PublicKey.X not match")
+}
+
+func TestParseCertificateNotMatchCurve(t *testing.T) {
+	assert.Contains(t, errorFrom(util.CheckCertificate(pkgt.B3Certs[0],
+		pkgt.B3KeyP521, "amppackageexample.com")), "PublicKey.Curve not match")
+}
+
+func TestParseCertificateNotMatchDomain(t *testing.T) {
+	assert.Contains(t, errorFrom(util.CheckCertificate(pkgt.B3Certs2[0],
+		pkgt.B3Key2, "amppackageexample.com")), "x509: certificate is valid for amppackageexample2.com, www.amppackageexample2.com, not amppackageexample.com")
+}
+
+func TestParse91DaysCertificate(t *testing.T) {
+	assert.Contains(t, errorFrom(util.CheckCertificate(pkgt.B3Certs91Days[0],
+		pkgt.B3Key, "amppackageexample.com")), "Certificate MUST have a Validity Period no greater than 90 days")
+}
+
+func TestParseCertificateIssuedBeforeMay1(t *testing.T) {
+	aug1st := time.Date(2009, time.August, 1, 0, 0, 0, 0, time.UTC)
+	if time.Now().After(aug1st) {
```

Comment:
Is it possible to make this test stable in the face of an ever-changing `time.Now()`? I suggest one of two options:

 - Pass `now time.Date` as another param to `util.CheckCertificate`.
 - Temporarily modify `start90DayGracePeriod` and `end90DayGracePeriod` (introduced per my other suggestion), to elicit the desired test conditions.

The benefit is that both cases are tested now -- we don't have a surprise test failure on 1 Aug. (I trust that you tested the Aug 1 case, but don't trust that future changes will.)

I have a slight preference for the first option, seems simpler, but not a strong preference.

Reply:
fixed with the first option.